### PR TITLE
topsql: do not aggregate when topsql is not enabled

### PR DESCRIPTION
**Pull request by mornyx**
Signed-off-by: mornyx <mornyx.z@gmail.com>

Issue Number: ref #31704

### Release note

```release-note
None
```

### CODE REVIEW

**Review by breeswish — APPROVED** [review]

**Comment by breeswish** [comment]
Could you also supply a test to ensure that when top sql is not enabled, the aggregation will not do?

**Review by crazycs520 — APPROVED** [review]

**Comment by mornyx** [comment]
> Could you also supply a test to ensure that when top sql is not enabled, the aggregation will not do?

done

**Review by breeswish — APPROVED** [review]

**Comment by breeswish** [comment]
When this PR is merged, I will open a new issue over #31704 specifically describe the memory leaking.

File: util/topsql/stmtstats/aggregator.go [hunk 1]
```diff
@@ -19,9 +19,12 @@ import (
 	"sync"
 	"time"
 
+	"github.com/pingcap/tidb/util/topsql/state"
 	"go.uber.org/atomic"
 )
 
+const maxStmtStatsSize = 1000000
```
Comment (mornyx) [inline util/topsql/stmtstats/aggregator.go:26]: @crazycs520 Please help me to confirm whether this value is enough to protect TiDB.

**Comment by hawkingrei** [comment]
It have data race problem in the test #31713

**Comment by mornyx** [comment]
> It have data race problem in the test #31713

Fixed in https://github.com/pingcap/tidb/pull/31714

**Pull request merged**